src/instructlab/model/upload.py

```diff
@@ -276,3 +298,91 @@ def upload_oci(self) -> None:
         logger.info(
             f"Uploading OCI model at {self.local_model_path} succeeded!",
         )
+
+
+class S3ModelUploader(ModelUploader):
+    """Class to handle uploading models to S3"""
+
+    def __init__(
+        self,
+        model: str,
+        destination: str,
+    ) -> None:
+        super().__init__(
+            model=model,
+            destination=destination,
+        )
+
+    def upload(self) -> None:
+        """
+        Upload specified model to the specified S3 endpoint
+        """
+
+        # AWS cred check
+        try:
+            sts_client = boto3.client("sts")
+            sts_client.get_caller_identity()
+        except Exception as exc:
+            raise ValueError(
+                """AWS credentials need to be set in your environment to upload to S3.
+                You can set your credentials using the AWS CLI
+                For alternative methods to set credentials, please review the boto3 documentation"""
+            ) from exc
+
+        logger.info(
+            f"Uploading model to S3 endpoint:\n{DEFAULT_INDENT}Model: {self.model}\n{DEFAULT_INDENT}Destination: {self.destination}\n"
+        )
+
+        # determine if model is path or name - look in checkpoints dir if the latter
+        self.is_model_path_or_name()
+
+        # raise exception if user did not provide valid S3 bucket endpoint
+        if not is_s3_repo(self.destination):
+            logger.error(
+                f"\nInvalid destination supplied:\n{DEFAULT_INDENT}Please specify valid S3 bucket URL syntax via --destination",
```

Comment:
minor nitpick for clarity. (sometimes when there are a lot of logs printed out, it can be difficult for users to debug issues.)

```suggestion
                f"\nInvalid S3 destination supplied:\n{DEFAULT_INDENT}Please specify valid S3 bucket URL syntax via --destination",
```

Reply:
Done!